rustls/src/tls_client_connection.rs

```diff
@@ -33,9 +33,9 @@ use TlsClientStreamBuilder;
 ///
 /// Use with `trust_dns::client::Client` impls
 pub struct TlsClientConnection {
-    io_loop: Core,
-    tls_client_stream: Box<Future<Item = TlsClientStream, Error = io::Error>>,
-    client_stream_handle: Box<DnsStreamHandle<Error = ClientError>>,
+    builder: TlsClientStreamBuilder,
+    name_server: SocketAddr,
+    subject_name: String,
```

Comment:
This should probably be named `dns_name`. Will save confusion about whether it is the subject CN (it's not).

Reply:
I’m not sure exactly what you mean. This must match the subject name in the certificate returned by the DNS server. I also have been thinking of adding SNI support on the server such that a nameserver acting as an authority for multiple zones could return different certificates based on this.

Can you clarify what you mean?